lib/intacct/models/sotransaction.rb

```diff
@@ -335,78 +338,80 @@ def create_xml(xml)
         end
       end
 
-      def update_xml(xml)
-        if attributes.datecreated.present?
+      def update_xml(xml, options = {})
+        update_attributes = options.blank? ? attributes : OpenStruct.new(options)
```

Comment:
I think the name `options` is confusing since it is used to pass attributes. Maybe `attributes_override`?